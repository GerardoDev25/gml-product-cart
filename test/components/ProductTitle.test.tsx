import React from 'react';
import rendered from 'react-test-renderer';
import { ProductTitle, ProductCard } from '../../src/components';
import { product1 } from './data/product';

describe('test in productTitle', () => {
  test('should to show the component correctly with the title costumed', () => {
    const wrapper = rendered.create(
      <ProductTitle title="custom product" className="custome-class" />
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should to show the component with the product title', () => {
    const wrapper = rendered.create(
      <ProductCard product={product1}>{() => <ProductTitle />}</ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
