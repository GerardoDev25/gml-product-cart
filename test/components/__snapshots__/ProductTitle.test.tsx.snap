// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in productTitle should to show the component correctly with the title costumed 1`] = `
<span
  className="productDescription custome-class"
>
  custom product
</span>
`;

exports[`test in productTitle should to show the component with the product title 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - sin IMG
  </span>
</div>
`;
